src/core/string_map.cc

```diff
@@ -142,6 +162,8 @@ void StringMap::RandomPairsUnique(unsigned int count, std::vector<sds>& keys,
 
 void StringMap::RandomPairs(unsigned int count, std::vector<sds>& keys, std::vector<sds>& vals,
                             bool with_value) {
+  CollectExpired();
```

Comment:
maybe add a comment, this is `O(n)` function that may slow down `RandomPairs` for large sets?